### ISSUE TITLE
cuda.cmake: Found Conflicting, Identical CUDA Install

### ISSUE THREAD

**Issue by ax3l** [pytorch/pytorch#108931]
### 🐛 Describe the bug

Hi,

I am compiling pytorch 2.1.0-rc3 from source on RHEL8 using the PPC64LE CPU architecture and CUDA support (7.0 for V100).

During compilation, somehow the following configure error occurs:
```
CMake Error at cmake/public/cuda.cmake:65 (message):
  Found two conflicting CUDA installs:

  V12.0.76 in '/usr/tce/packages/cuda/cuda-12.0.0/include' and

  V12.0.76 in '/usr/tce/packages/cuda/cuda-12.0.0/nvidia/include'
Call Stack (most recent call first):
  cmake/Dependencies.cmake:44 (include)
  CMakeLists.txt:722 (include)
```

Should this error be skipped if the same version is identified?
https://github.com/pytorch/pytorch/blob/v2.1.0-rc3/cmake/public/cuda.cmake#L63-L68

### Versions

- pytorch 2.1.0-rc3
- CUDA 12.0.76
- Clang 12.0.1
- RHEL 8.7

cc @malfet @seemethere @ptrblck

**Comment by ax3l** [comment]
Tested further: the same problem did not appear with PyTorch 2.0.1.

**Comment by ax3l** [comment]
The directories are truly identical:
```
$ ls -hal /usr/tce/packages/cuda/cuda-12.0.0/include
lrwxrwxrwx 1 gyllen gyllen 14 Dec 12  2022 /usr/tce/packages/cuda/cuda-12.0.0/include -> nvidia/include
```

**Comment by ax3l** [comment]
Found and submitted a fix in #108932

**Comment by anthonyalayo** [comment]
I have the same problem.
